src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/DesignTimeTargets/Microsoft.CSharp.DesignTime.targets

```diff
@@ -64,6 +64,41 @@
 
   <!-- Targets -->
 
+  <!-- Razor cohosting support for projects that don't use the Razor SDK -->
+
+  <PropertyGroup>
+    <_HasRazorOrCshtmlFiles>false</_HasRazorOrCshtmlFiles>
+  </PropertyGroup>
+
+  <!-- .NET Framework uses a different editor in VS -->
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework' And 'UsingMicrosoftNETSdkRazor' != 'true'">
+    <!-- Collect all Razor and CSHTML files -->
+    <_RazorOrCshtmlFiles Include="**\*.razor;**\*.cshtml" />
+  </ItemGroup>
+
+  <Target Name="DetectRazorOrCshtmlFiles" BeforeTargets="AssignProjectConfiguration" Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' And 'UsingMicrosoftNETSdkRazor' != 'true'">
+    <PropertyGroup Condition="'@(_RazorOrCshtmlFiles)' != ''">
+      <_HasRazorOrCshtmlFiles>true</_HasRazorOrCshtmlFiles>
+    </PropertyGroup>
+  </Target>
```

Comment:
I'm surprised this is needed. I wouldn't think it'd help either, given the stuff below here is evaluated and this property would be set too late (if I'm remembering my MSBuild ordering correctly.)